lib/driver.js

```diff
@@ -562,10 +562,13 @@ class XCUITestDriver extends BaseDriver {
         await markSystemFilesForCleanup(this.wda);
       }
 
-      // TODO: platformVersion should be a required cap
-      if (this.isSafari() && util.compareVersions(this.opts.platformVersion, '>=', '12.2')) {
-        // on 12.2 the page is not opened in WDA
-        await this.setUrl(this._currentUrl);
+      if (this.isSafari() && !this.isRealDevice()) {
+        // on real devices this often causes other things to open
+        // TODO: platformVersion should be a required cap
+        if (util.compareVersions(this.opts.platformVersion, '>=', '12.2')) {
```

Comment:
there's no need for nested if. We could combine the condition with `&&`

Reply:
I'd rather nest when conditions are entirely different.